tests/sentry/integrations/slack/test_client.py

```diff
@@ -0,0 +1,128 @@
+        self.access_token = "xoxb-access-token"
+        self.integration = self.create_integration(
+            organization=self.organization,
+            external_id="slack:1",
+            provider="slack",
+            metadata={"access_token": self.access_token},
+        )
+        self.organization_integration = OrganizationIntegration.objects.get(
+            integration_id=self.integration.id
+        )
+        self.payload = {"channel": "#announcements", "message": "i'm ooo next week"}
+        self.mock_user_access_token_response = {"ok": True, "auth": "user"}
+        self.mock_access_token_response = {"ok": True, "auth": "token"}
+        self.mock_not_authed_response = {"ok": True, "auth": None}
+        base_response_kwargs = {
+            "method": responses.POST,
+            "url": re.compile(r"\S+chat.postMessage$"),
+            "status": 200,
+            "content_type": "application/json",
+        }
+        responses.add(
+            **base_response_kwargs,
+            json=self.mock_user_access_token_response,
+            match=[
+                matchers.header_matcher(
+                    {"Authorization": f"Bearer {self.user_access_token}"},
+                )
+            ],
+        )
+        responses.add(
+            **base_response_kwargs,
+            json=self.mock_access_token_response,
+            match=[matchers.header_matcher({"Authorization": f"Bearer {self.access_token}"})],
+        )
+        responses.add(
+            **base_response_kwargs,
+            json=self.mock_not_authed_response,
+            match=[matchers.header_matcher({})],
+        )
+
+    @responses.activate
+    def test_integration_proxy_is_active(self):
+        class SlackProxyTestClient(SlackClient):
+            use_proxy_url_for_tests = True
+
+        with override_settings(SILO_MODE=SiloMode.MONOLITH):
+            client = SlackProxyTestClient(integration_id=self.integration.id)
+            client.post("/chat.postMessage", data=self.payload)
+            request_url = responses.calls[0].request.url
+
+            assert PROXY_BASE_PATH not in request_url
+            assert client.base_url in request_url
+
+        responses.calls.reset()
+        with override_settings(SILO_MODE=SiloMode.CONTROL):
+            client = SlackProxyTestClient(integration_id=self.integration.id)
+            client.post("/chat.postMessage", data=self.payload)
+            request_url = responses.calls[0].request.url
+
+            assert PROXY_BASE_PATH not in request_url
+            assert client.base_url in request_url
+
+        responses.calls.reset()
+        with override_settings(SILO_MODE=SiloMode.REGION):
+            client = SlackProxyTestClient(integration_id=self.integration.id)
+            client.post("/chat.postMessage", data=self.payload)
+            request_url = responses.calls[0].request.url
+
+            assert PROXY_BASE_PATH in request_url
+            assert client.base_url not in request_url
```

Comment:
Should we also have an assertion for the proxy_base_path being in the proxy headers?